src/integrations/kotlin/nomic/data/repositories/games/GameRepositoryTest.kt

```diff
@@ -0,0 +1,107 @@
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.MOCK
+)
+class GameRepositoryTest(@Autowired private val db: Database) {
+
+    private companion object {
+        private val user1: EndUser = EndUser(50, "ListGamesUser1")
+        private val user2: EndUser = EndUser(55, "ListGamesUser2")
+    }
+
+    @Test
+    fun test_listGames_dataFromDatabase() {
+        val repo = GameRepository(db)
+        val expectedGames = db.games.map { GameModel(it.gameId, it.title, it.createDate, it.currentPlayer, it.user.id) }
+            .associateBy { it.gameId }.toMutableMap()
+
+        val games = repo.listGames()
+
+        for (game in games) {
+            Assertions.assertThat(expectedGames).containsKey(game.gameId)
+            val expectedGame = expectedGames[game.gameId]!!
+
+            Assertions.assertThat(game.createDate).isEqualTo(expectedGame.createDate)
+            Assertions.assertThat(game.userId).isEqualTo(expectedGame.userId)
+            Assertions.assertThat(game.title).isEqualTo(expectedGame.title)
+            Assertions.assertThat(game.currentPlayer).isEqualTo(expectedGame.currentPlayer)
+
+            expectedGames.remove(game.gameId)
+        }
+
+        Assertions.assertThat(expectedGames.size).isZero()
+    }
+
+    @Test
+    fun test_listGames_callsSpecificationsInOrder() {
+        val repo = GameRepository(db)
+        val specUser = IGameSpecification { games -> games.filter { it.userId eq 1601 } }
+
+        var hasFirstSpecRun = false
+
+        val firstSpec = IGameSpecification {
+                games ->
+            hasFirstSpecRun = true
+            games
+        }
+
+        val secondSpec = IGameSpecification {
+                games ->
+            Assertions.assertThat(hasFirstSpecRun).isTrue
+            games
+        }
+
+        val games = repo.listGames(specUser, firstSpec, secondSpec)
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = [1, 2, 4, 8, 16])
+    fun test_listGames_callsAllSpecifications(specificationCount: Int) {
+        val repo = GameRepository(db)
+        val mockSpecifications = mutableListOf<IGameSpecification>()
+        var totalSpecificationCalls = 0
+
+        for (i in 1..specificationCount) {
+            mockSpecifications.add(
+                mock<IGameSpecification> {
+                    on(it.apply(any<EntitySequence<GameDTO, Games>>())) doAnswer {
+                            answer ->
+                        totalSpecificationCalls++
+                        answer.getArgument(0)
+                    }
+                }
+            )
+        }
+
+        // Filter for the performance of the test so that the repo isn't processing a bunch of entities
+        val filterByUser = IGameSpecification { games -> games.filter { it.userId eq 1601 } }
+
+        repo.listGames(filterByUser, *mockSpecifications.toTypedArray())
+
+        Assertions.assertThat(totalSpecificationCalls).isEqualTo(specificationCount)
+    }
```

Comment:
Similar Idea as before
Gonna need an explanation on this test.
What is is doing?
What is the thing you are testing for?
Why are you testing this?